### PR TITLE
Introduce daemonbase and use it for logging.

### PR DESCRIPTION
**Pull request by partim**
This PR switches the logging configuration to use the new _daemonbase_ crate.

### CODE REVIEW

**Comment by ximon18** [comment]
Is it wise to make `main` depend on a Git branch of `daemonbase`? That would block you from releasing until `daemonbase` itself is released to crates.io, right?

**Comment by partim** [comment]
A _daemonbase_ release should probably happen before merging, indeed.

File: src/log.rs [hunk 1]
```diff
@@ -1,458 +0,0 @@
...
-    /// This is the non-Unix version that does not use syslog.
-    #[cfg(not(unix))]
-    #[allow(clippy::unnecessary_wraps)]
-    fn apply_log_matches(
-        &mut self,
-        matches: &ArgMatches,
-        cur_dir: &Path,
-    ) -> Result<(), Failed> {
-        if let Some(file) = matches.value_of("logfile") {
-            if file == "-" {
```
Comment (ximon18) [inline src/log.rs:143]: At a glance I didn't see support for "-" in daemonbase as a file path causing stderr logging to be used instead.

Comment (ximon18) [reply src/log.rs:143]: And trying `--logfile -` causes a `-` file to be created in my local directory.

Comment (partim) [reply src/log.rs:143]: This should indeed be fixed in _daemonbase._

Comment (partim) [reply src/log.rs:143]: This is addressed in https://github.com/NLnetLabs/daemonbase/pull/1

File: src/lib.rs [hunk 2]
```diff
@@ -27,7 +27,7 @@ pub mod comms;
 pub mod config;
 pub mod formats;
 pub mod http;
-pub mod log;
+//pub mod log;
```
Comment (ximon18) [inline src/lib.rs:30]: Delete this if it is no longer needed.
```suggestion
```

**Review by ximon18 — APPROVED** [review]

File: src/config.rs [hunk 3]
```diff
@@ -91,30 +78,42 @@ impl Config {
...
     /// configuration.
     pub fn from_arg_matches(
-        matches: &ArgMatches,
-        cur_dir: &Path,
+        matches: &clap::ArgMatches,
         manager: &mut Manager,
     ) -> Result<Self, Failed> {
-        let conf_path = cur_dir.join(matches.value_of("config").unwrap());
-        let conf = match ConfigFile::load(&conf_path) {
+        let args = Args::from_arg_matches(
```
[the remark this answers is not in the document]
Comment (partim) [reply src/config.rs:84]: expansion of relative paths is now done transparently in _daemonbase._

**Pull request merged**